sys/src/9/amd64/archamd64.c

```diff
@@ -67,16 +67,43 @@ cpuidinfo(uint32_t eax, uint32_t ecx, uint32_t info[4])
 	return 1;
 }
 
+static char *
+cpuidname(uint32_t *info0)
+{
+	char *vendorid;
+
+	if(memcmp(&info0[1], "GenuntelineI", 12) == 0) {
+		vendorid = "GenuineIntel";
+		return vendorid;
+	}
+	else if(memcmp(&info0[1], "AuthcAMDenti", 12) == 0) {
+		vendorid = "AuthenticAMD";
+		return vendorid;
+	}
+	else {
+		vendorid = nil;
+		panic("Panic!: unrecognized cpu %s\n", vendorid);
+	}
+	return vendorid;
+
+
+}
+
 static int64_t
 cpuidhz(uint32_t *info0, uint32_t *info1)
 {
 	int f, r;
 	int64_t hz;
 	uint64_t msr;
+	char *vendorid;
+
+	vendorid = cpuidname(info0);
 
-	//print("CPUID Vendor: %s\n", (char *)&info0[1]);
-	//print("CPUID Signature: %d\n", info1[0]);
-	if(memcmp(&info0[1], "GenuntelineI", 12) == 0){
+	print("CPUID Vendor: %s\n", (char *)&info0[1]);
```

Comment:
That are traces. They must be cleaned when WIP in tittle will be gone. Also, the isintelcpu useless check and element for mach struct (not used across all the kernel, I think it's and old jmk's test for debugging or anything) is commented now, but I'll remove it when this PR was working for all of us.